### DISCUSSION TITLE
Call method when ObservableProperty changes using CommunityToolkit.Mvvm

### DISCUSSION THREAD

**Discussion by imsam67** [CommunityToolkit/dotnet#338] [General]
I'm implementing auto complete feature in my .NET MAUI app and I'm using `CommunityToolkit.Mvvm` code generators in my view model to handle observable properties.

I have the following code and I'm trying call `GetSuggestions()` method when the `SearchText` changes.

```
    [ObservableProperty]
    [NotifyCanExecuteChangedFor(nameof(GetSuggestions))]
    string searchText;

    [ObservableProperty]
    bool showSuggestions;
    
    ObservableCollection<string> Suggestions { get; } = new();
    
    private async Task GetSuggestions()
    {
       if(string.IsNullOrEmpty(SearchText) || SearchText.Length < 3)
          return;
    
       var data = await _myApiService.GetSuggestions(SearchText.Trim());
       if(data != null && data.Count > 0)
       {
          Suggestions.Clear();
          foreach(var item in data)
             Suggestions.Add(item);

          ShowSuggestions = true;
       }
    }
```

This is giving me the following error:

> The target(s) of [NotifyCanExecuteChangedFor] must be an accessible IRelayCommand property, but "GetSuggestions" has no matches in type MyViewModel.

What am I doing wrong here?

**Comment by Balkoth** [comment]
You can not use `NotifyCanExecuteChangedFor` for this task. Currently you have to use the partial methods described here https://docs.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#running-code-upon-changes

Now everywhere you use this and do not use the `value` parameter you get warning `IDE0060 // Remove unused parameter` unless you disable them at the specific location, because they are meaningless there and you can do nothing about it.

IMHO there should be an easy way to attribute the property to call a method when the property has changed.

P.S.: Is the `value` parameter even needed? Does the backing field in the viewmodel at the time of the method call not always have the same value?
